### PR TITLE
🐛  Fix helm chart syntax errors

### PR DESCRIPTION
**Pull request by willie-yao**
**What this PR does / why we need it**:
This PR fixes a few edge cases that cause syntax errors in helm charts.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes part of #532

### CODE REVIEW

File: hack/charts/cluster-api-operator/templates/core.yaml [hunk 1]
```diff
@@ -45,7 +45,7 @@ spec:
 {{- end }}
 {{- if $.Values.manager }}
   manager:
-{{- if $.Values.manager.featureGates.core }}
+{{- if and $.Values.manager.featureGates $.Values.manager.featureGates.core }}
```
Comment (willie-yao) [inline hack/charts/cluster-api-operator/templates/core.yaml:48]: Fixes nil pointer exception if manager is defined but not manager.featureGates

File: hack/charts/cluster-api-operator/templates/infra.yaml [hunk 2]
```diff
@@ -38,15 +38,15 @@ metadata:
   annotations:
     "helm.sh/hook": "post-install"
     "helm.sh/hook-weight": "2"
-{{- if or $infrastructureVersion $.Values.configSecret.name }}
+{{- if or $infrastructureVersion $.Values.configSecret.name $.Values.manager }}
```
Comment (willie-yao) [inline hack/charts/cluster-api-operator/templates/infra.yaml:41]: This fixes a bug where `spec:` is not printed when infrastructureVersion .Values.configSecret.name are not specified and Values.manager is.

File: hack/charts/cluster-api-operator/templates/infra.yaml [hunk 3]
```diff
@@ -38,15 +38,15 @@ metadata:
...
 spec:
 {{- end }}
 {{- if $infrastructureVersion }}
   version: {{ $infrastructureVersion }}
 {{- end }}
 {{- if $.Values.manager }}
   manager:
-{{- if hasKey $.Values.manager.featureGates $infrastructureName }}
+{{- if and (kindIs "map" $.Values.manager.featureGates) (hasKey $.Values.manager.featureGates $infrastructureName) }}
```
Comment (willie-yao) [inline hack/charts/cluster-api-operator/templates/infra.yaml:49]: This fixes a type check error if $.Values.manager.featureGates is not specified.

**Review by Fedosin — APPROVED** [review]
/lgtm

**Review by mboersma — APPROVED** [review]
/lgtm

**Review by alexander-demicev — APPROVED** [review]
thanks a lot for fixing this

/approve

**Pull request merged**